fastly/products/bot_management/api.go

```diff
@@ -0,0 +1,38 @@
+package bot_management
+
+import (
+	"github.com/fastly/go-fastly/v9/fastly"
+	// fp is 'fastly products' package
+	fp "github.com/fastly/go-fastly/v9/fastly/products"
+	// ip is 'internal products' package
+	ip "github.com/fastly/go-fastly/v9/internal/products"
+)
+
+const ProductID = "bot_management"
+
+// Get gets the status of the Bot Management product on the service.
+func Get(c *fastly.Client, serviceID string) (*fp.EnableOutput, error) {
+	return ip.Get(&ip.GetInput[fp.EnableOutput]{
+		Client:    c,
+		ProductID: ProductID,
+		ServiceID: serviceID,
+	})
+}
```

Comment:
`GetInput` takes the type for just one reason: so that type inference can supply it to `Get`. This was done for consistency with the other operations that *do* have an `Input` type, so that we don't have type parameters on the function in some places and on the struct in other places. I explained this in some of the godocs as well, if those explanations aren't sufficient I'm happy to expand/reword them.

As far as the package naming... I'm not tied to any particular model, but I really don't want to overload 'API' like this. Maybe `productcore` or something like that would work.